source/dofs/dof_handler.cc

```diff
@@ -855,7 +855,7 @@ DoFHandler<dim, spacedim>::DoFHandler(const Triangulation<dim, spacedim> &tria)
                                ParallelShared<DoFHandler<dim, spacedim>>>(
         *this);
   else if (dynamic_cast<
-             const parallel::distributed::Triangulation<dim, spacedim> *>(
+             const parallel::DistributedTriangulationBase<dim, spacedim> *>(
```

Comment:
I suspect that there are going to be many cases like this where we'll have to change the dynamic-cast. If you want, make this into a separate PR -- it's obviously correct -- and I'll approve this separately.

Reply:
It is quite annoying that there is not `else Assert(...)`. If this would have been there, I would have noticed my mistake a month ago... I think we have to look through all `dynamic_cast`s...

Reply:
Yes :-(

Reply:
I would like to keep the bug-fixes of this PR together since they depend on each other (for example, one test crashed with these changes and was fixed after I have introduced the `catch` - see below).